File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_04504.py
# SPDX-License-Identifier: Apache-2.0
from flask import request, jsonify


def BenchmarkTest04504():
    auth_header = request.headers.get('Authorization', '')
    data = auth_header if auth_header else 'default'
    if str(data) == 'S3cr3tToken':
        return jsonify({'authenticated': True}), 200
    return jsonify({"result": "success"})

File: Benchmarks/quicktest/java/spring/testcode/BenchmarkTest04504.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest04504 {

    @GetMapping("/BenchmarkTest04504")
    public void BenchmarkTest04504(@RequestHeader("X-Custom-Header") String xCustomHeader, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String headerValue = xCustomHeader != null ? xCustomHeader : "";
        java.util.Map.Entry<String,String> edge = java.util.Map.entry(headerValue, "http");
        response.setHeader("X-Tuple-Context", edge.getValue());
        String data = edge.getKey();
        response.getWriter().print("<div>" + data + "</div>");
    }
}
